src/classes/AFFL_AccChange_TDTM.cls

```diff
@@ -56,93 +61,163 @@ public with sharing class AFFL_AccChange_TDTM extends TDTM_Runnable {
+        dmlWrapper = null;
+  
+        if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {      
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_AccChange_TDTM, false);
             TDTM_ProcessControl.resetRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update);
         }
         return dmlWrapper;
     }
+   
+    /******************************************************************************************************************************
+    * @description Creates a map Affiliation Id to other Primary Affilaitons with same Account and Contact
+    * @param newAffiliationList List of affiliations that are being updated
+    * @return idToOtherAffiliationWithSameAccAndCon A map Affiliation Id to other Primary Affilaitons with same Account and Contact
+    ********************************************************************************************************************************/
+    @testVisible
+    private Map<Id, List<Affiliation__c>> createdIdToOtherPrimaryAfflsWithSameAccAndCon(List<Affiliation__c> newAffiliationList) {
+
+        Set<ID> newPrimaryAfflSet = new Set<ID>();
+        Set<ID> afflContactIdSet = new Set<ID>();
+        Set<String> afflAccIdSet = new Set<String>();
+
+        for (Affiliation__c affl : newAffiliationList) {
+            newPrimaryAfflSet.add(affl.ID);
+            afflContactIdSet.add(affl.Contact__c);
+            afflAccIdSet.add(affl.Account__c);
+        }
+        
+        Map<String, List<Affiliation__c>> idToOtherPrimaryAfflList = new Map<String, List<Affiliation__c>>();
+        
+        // Get all possible Primary Affiliation with Same Contact and Account
+        for (Affiliation__c affiliation : [SELECT Id, Contact__c, Account__c, Affiliation_Type__c, Primary__c
+                                            FROM Affiliation__c WHERE Account__c IN :afflAccIdSet 
+                                            AND Contact__c IN :afflContactIdSet
+                                            AND Primary__c = TRUE 
+                                            AND Id NOT IN :newPrimaryAfflSet]) {
+            string accIdConId = (String)affiliation.Account__c + (String)affiliation.Contact__c;                              
+            if (idToOtherPrimaryAfflList.containsKey(accIdConId)) {
+                idToOtherPrimaryAfflList.get(accIdConId).add(affiliation);
+            } else {
+                idToOtherPrimaryAfflList.put(accIdConId, new List<Affiliation__c>{affiliation});
+             }
+        }
+
+        Map<Id, List<Affiliation__c>> idToOtherAffiliationWithSameAccAndCon = new Map<Id, List<Affiliation__c>>();
+        
+        for (Affiliation__c affiliation: newAffiliationList) {
+            string accIdConId = (String)affiliation.Account__c + (String)affiliation.Contact__c;
+            if (idToOtherPrimaryAfflList.isEmpty() == false && idToOtherPrimaryAfflList.get(accIdConId).isEmpty() == false) {
```

Comment:
```suggestion
            if (idToOtherPrimaryAfflList.get(accIdConId) != null && idToOtherPrimaryAfflList.get(accIdConId).isEmpty() == false) {
```
In order to be certain we don't get a null pointer exception, we need to verify that the list retrieved from the map is not null before calling one of its methods.